py/tools/venv_bin/src/main.rs

```diff
@@ -115,76 +115,65 @@ struct VenvArgs {
 
 fn venv_cmd_handler(args: VenvArgs) -> miette::Result<()> {
     let pth_file = py::PthFile::new(&args.pth_file, args.pth_entry_prefix);
-    match args.mode {
+    if let VenvMode::DynamicSymlink = args.mode {
         // FIXME: Does this need to care about the repo?
-        VenvMode::DynamicSymlink => py::create_venv(
+        return py::create_venv(
             &args.python,
             &args.location,
             Some(pth_file),
             args.collision_strategy.unwrap_or_default().into(),
             &args.venv_name,
-        ),
+        );
+    }
 
-        it => {
-            let Some(version) = args.version else {
-                return Err(miette!("Version must be provided for static venv modes"));
+    let version = args
+        .version
+        .ok_or_else(|| miette!("Version must be provided for static venv modes"))?;
+
+    let venv = py::venv::create_empty_venv(
+        args.repo
+            .as_deref()
+            .ok_or_else(|| miette!("The --repo argument is required for static venvs!"))?,
+        &args.python,
+        py::venv::PythonVersionInfo::from_str(&version)?,
+        &args.location,
+        args.env_file.as_deref(),
+        args.venv_shim.as_deref(),
+        args.debug,
+        args.include_system_site_packages,
+        args.include_user_site_packages,
+    )?;
+
+    let strat: Box<dyn PthEntryHandler> = match args.mode {
```

Comment:
you have to annotate the virtual type here, because if you don't, the match will type-hint off the first real type it returns. The _first_ arm types as `!`, which isn't real. The second arm is `Box<py::venv::PthStrategy>`, which means all other arms need to type as that concrete instance. `!` types that way, but `Box<py::venv::FirstPartyThirdPartyStrategy>` does not.

You can dodge this by typing the binding site (`let strat: ...`) or by as-casting the box. `Box::new(...) as Box<dyn ...>` is more annoying, imo